src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -261,6 +292,10 @@ public void visitToken(DetailAST ast) {
         if (isInterfaceOrAnnotationMember(ast)) {
             processInterfaceOrAnnotation(ast);
         }
+
+        if (jdkVersion >= JAVA_17) {
+            checkForRedundantModifier(ast, TokenTypes.STRICTFP);
```

Comment:
This is breaking behavior compatibility.
I am ok to make default to be most recent version because this modifier is super rear in use.

Please put in issue example on violation message that users might get after update upgrade to latest checkstyle and what change to config they should do. Let's presume they should put `11` in this property to bring behavior back to previous.

Reply:
>I am ok to make default to be most recent version because this modifier is super rear in use.

How can we keep this updated? Ideally we would have some source of truth in this repo that declares what java language version we support up to, which is then referenced by some test for this, our docs, etc.

Reply:
I don't know.

Reply:
We keep this updated by passing [`java.version`](https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L224) to the Checkstyle run and pass this to the property of the check.

I think this would also solidify the other discussion that we must follow the pattern of this field. [In old versions, it was `1.8` for Java 8.](https://github.com/rnveach/checkstyle-backport-jre8/blob/master/pom.xml#L79) I don't know if it supported `8` instead.

Reply:
`java.version` won't be the latest version of java that we support, though. Either way, we can discuss this elsewhere, let's create a ticket if we want to pursue this.

Reply:
We can do


>make default to be most recent version because this modifier is super rear in use.

So - Java 21 will be our default since this is the highest version that we support by language level

Reply:
> How can we keep this updated?

how about documenting that
in the check update doc? mention that some checks have `be considered` property these should be consider to be updated

Reply:
We still need documentation on what versions  this field can take. Kind of an enumeration.
Versions like 1.8 does not make sense to support as before "17" all was compatible.
We can just explain in property description that we support versions 11,12,13,14,15,16,17,... and up.

Do we have example that users can get value `11` from build system?

Please update checkstyle config to use this property, it will be example for ANT :).

We have to eat food we produce.

Reply:
> java.version won't be the latest version of java that we support, though

I thought we were talking about connecting it to our (CS) version, not for users.

Reply:
> We keep this updated by passing [java.version](https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L224) to the Checkstyle run and pass this to the property of the check.


done